File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
from bisect import bisect_left, bisect_right
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        x = bisect_left(nums,target)
        if x == len(nums) or nums[x] != target: return [-1,-1]
        y = x
        while y<len(nums) and nums[y] == target: y+=1
        return [x,y-1]

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int x = lower_bound(nums.begin(), nums.end(), target) - nums.begin();
        if (x >= nums.size() || nums[x] != target) return {-1,-1};
        int y = x;
        while(y<nums.size() && nums[y] == target)y++;
        return {x,y-1};
    }
};